tests/release/service/release_plan_and_admission_matched.go

```diff
@@ -188,9 +187,11 @@ var _ = framework.ReleaseServiceSuiteDescribe("ReleasePlan and ReleasePlanAdmiss
 					if condition == nil {
 						return fmt.Errorf("failed to get the MatchedConditon of %s", secondReleasePlanCR.Name)
 					}
+					if condition.Status == metav1.ConditionTrue {
+						return fmt.Errorf("the MatchedConditon of %s has not reconciled yet", secondReleasePlanCR.Name)
+					}
 					return nil
 				}, releasecommon.ReleasePlanStatusUpdateTimeout, releasecommon.DefaultInterval).Should(Succeed())
-				Expect(secondReleasePlanCR.IsMatched()).To(BeFalse())
```

Comment:
Why is this line being removed?

Reply:
The checking is replaced by the new added "if" statement in the loop. 
 _if condition.Status == metav1.ConditionTrue {
						return fmt.Errorf("the MatchedConditon of %s has not reconciled yet", secondReleasePlanCR.Name)_

Reply:
Do you think this double checking is still needed?

Reply:
Ah! I missed that. Clever. Nope, we don't need a double check. I think a comment on line 190 like what you explained here could be nice, but not required

Reply:
OK. Let me add a comment.  Thanks